myENA/cloudstack
# EC-3970 - Change listVpcOfferings endpoint and create a vpc offering

## Description

Currently in cloudstack there is no way to create a VPC Offering with a selected service offering, the service offering is always null. When a VPC router is created if the service offering is null a default Service Offering is chosen.

The following changes have been added:
1. listVpcOfferings endpoint will list what system service offering they were created with
2. the cloudstack ui will let you pick what serviceOfferingId to use








## Types of changes


## Screenshots (if appropriate):
![vpcOffering](https://user-images.githubusercontent.com/50488048/74658782-aa781a80-519b-11ea-96fd-6996f7cb975e.PNG)

## How Has This Been Tested?

Wrote a python test for testing create VPC offering with selected service offering
Steps for validation
1. Create a system offering
2. Create VPC Offering by specifying all supported Services and your system offering
3. VPC offering should be created successfully

[vpcOfferingWithSelectedServiceOfferingTestLogs.txt](https://github.com/myENA/cloudstack/files/4219235/vpcOfferingWithSelectedServiceOfferingTestLogs.txt)
